### ISSUE TITLE
[BUG] cmake output complains about missing export.h

### ISSUE THREAD

**Issue by yurivict** [google/benchmark#1454]
```
-- Performing Test HAVE_THREAD_SAFETY_ATTRIBUTES
-- Performing Test HAVE_THREAD_SAFETY_ATTRIBUTES
-- Performing Test HAVE_THREAD_SAFETY_ATTRIBUTES -- failed to compile: Change Dir: /disk-samsung/freebsd-ports/devel/benchmark/work/.build/CMakeFiles/CMakeTmp

Run Build Command(s):/usr/local/bin/ninja cmTC_a4164 && [1/2] Building CXX object CMakeFiles/cmTC_a4164.dir/thread_safety_attributes.cpp.o
[2/2] Linking CXX executable cmTC_a4164
FAILED: cmTC_a4164 
: && /usr/bin/c++ -O2 -pipe -fno-omit-frame-pointer -fstack-protector-strong -fno-strict-aliasing -fno-omit-frame-pointer   -I/disk-samsung/freebsd-ports/devel/benchmark/work/benchmark-1.7.0/include  -std=c++11  -Wall  -Wextra  -Wshadow  -Wfloat-equal  -Wsuggest-override  -pedantic  -pedantic-errors  -Wshorten-64-to-32  -fstrict-aliasing  -Wno-deprecated-declarations  -Wstrict-aliasing  -Wthread-safety -fstack-protector-strong CMakeFiles/cmTC_a4164.dir/thread_safety_attributes.cpp.o -o cmTC_a4164   && :
ld: error: undefined symbol: benchmark::internal::InitializeStreams()
>>> referenced by thread_safety_attributes.cpp
>>>               CMakeFiles/cmTC_a4164.dir/thread_safety_attributes.cpp.o:(_GLOBAL__sub_I_thread_safety_attributes.cpp)
c++: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.


-- Performing Test HAVE_CXX_FLAG_COVERAGE
-- Performing Test HAVE_CXX_FLAG_COVERAGE - Success
-- Performing Test HAVE_STD_REGEX
-- Performing Test HAVE_STD_REGEX
-- Performing Test HAVE_STD_REGEX -- success
-- Performing Test HAVE_GNU_POSIX_REGEX
-- Performing Test HAVE_GNU_POSIX_REGEX
-- Performing Test HAVE_GNU_POSIX_REGEX -- failed to compile: Change Dir: /disk-samsung/freebsd-ports/devel/benchmark/work/.build/CMakeFiles/CMakeTmp

Run Build Command(s):/usr/local/bin/ninja cmTC_b787d && [1/2] Building CXX object CMakeFiles/cmTC_b787d.dir/gnu_posix_regex.cpp.o
FAILED: CMakeFiles/cmTC_b787d.dir/gnu_posix_regex.cpp.o 
/usr/bin/c++   -O2 -pipe -fno-omit-frame-pointer -fstack-protector-strong -fno-strict-aliasing -fno-omit-frame-pointer   -I/disk-samsung/freebsd-ports/devel/benchmark/work/benchmark-1.7.0/include  -std=c++11  -Wall  -Wextra  -Wshadow  -Wfloat-equal  -Wsuggest-override  -pedantic  -pedantic-errors  -Wshorten-64-to-32  -fstrict-aliasing  -Wno-deprecated-declarations  -Wstrict-aliasing  -Wthread-safety -MD -MT CMakeFiles/cmTC_b787d.dir/gnu_posix_regex.cpp.o -MF CMakeFiles/cmTC_b787d.dir/gnu_posix_regex.cpp.o.d -o CMakeFiles/cmTC_b787d.dir/gnu_posix_regex.cpp.o -c /disk-samsung/freebsd-ports/devel/benchmark/work/benchmark-1.7.0/cmake/gnu_posix_regex.cpp
/disk-samsung/freebsd-ports/devel/benchmark/work/benchmark-1.7.0/cmake/gnu_posix_regex.cpp:1:10: fatal error: 'gnuregex.h' file not found
#include <gnuregex.h>
         ^~~~~~~~~~~~
1 error generated.
ninja: build stopped: subcommand failed.


-- Performing Test HAVE_POSIX_REGEX
-- Performing Test HAVE_POSIX_REGEX
-- Performing Test HAVE_POSIX_REGEX -- success
-- Performing Test HAVE_STEADY_CLOCK
-- Performing Test HAVE_STEADY_CLOCK
-- Performing Test HAVE_STEADY_CLOCK -- success
-- Looking for C++ include pthread.h
-- Looking for C++ include pthread.h - found
-- Found Threads: TRUE  
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_C_COMPILER
    CMAKE_C_FLAGS
    CMAKE_C_FLAGS_DEBUG
    CMAKE_C_FLAGS_RELEASE
    CMAKE_MODULE_LINKER_FLAGS
    CMAKE_VERBOSE_MAKEFILE


-- Build files have been written to: /disk-samsung/freebsd-ports/devel/benchmark/work/.build
```

**Comment by dominichamon** [comment]
this is deliberate. one of the hardest to debug and most confusing issues is when a particular combination of toolchain and library fail all three of the regex checks in cmake.  by having the error output inline, we are able to debug the issue much quicker.

it could be disabled, but then if a user has the issue we first have to go back and ask them to enable this and rebuild so it saves a round trip of debugging.

**Comment by yurivict** [comment]
On other occasions it also prints missing headers errors for export.h
Is this also deliberate?

**Comment by dominichamon** [comment]
yes. that would be an issue as "export.h" is part of the public interface and it being missing will be a problem for any integrating project.

unless I misunderstood what that error is.

**Comment by yurivict** [comment]
"export.h"-related errors are printed from configure when benchmark isn't installed. configure fails to find its own headers and then continues and succeeds.

This sounds like a problem.

**Comment by dominichamon** [comment]
i agree.

**Comment by dominichamon** [comment]
i have a change to pass in the include path correctly, however now the linking is failing as we need the benchmark library to link against to create the executable.

given that we only need to compile and not run, i think this may be unnecessary.

**Comment by dominichamon** [comment]
i have a fix here.. it required removing some deps on benchmark.h but that's good, because they shouldn't have been there in the first place :)

**Comment by bdaase** [comment]
Are you sure that the problem is solved? I tried to build 974cd5a5c5a78e76ebc50961f4dbf3bf6d4ade4e today and am still getting

```
➜  build git:(master) ✗ cmake .. -DCMAKE_BUILD_TYPE=Release
-- The C compiler identification is GNU 12.1.1
-- The CXX compiler identification is GNU 12.1.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/lib64/ccache/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/lib64/ccache/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Python: /usr/bin/python3.10 (found version "3.10.6") found components: Interpreter 
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- LLVM FileCheck Found: /usr/bin/FileCheck
-- Found Git: /usr/bin/git (found version "2.37.1") 
-- git version: v1.7.0-8-gaf32e3fe normalized to 1.7.0.8
-- Version: 1.7.0.8
-- Looking for shm_open in rt
-- Looking for shm_open in rt - found
-- Performing Test HAVE_CXX_FLAG_STD_CXX11
-- Performing Test HAVE_CXX_FLAG_STD_CXX11 - Success
-- Performing Test HAVE_CXX_FLAG_WALL
-- Performing Test HAVE_CXX_FLAG_WALL - Success
-- Performing Test HAVE_CXX_FLAG_WEXTRA
-- Performing Test HAVE_CXX_FLAG_WEXTRA - Success
-- Performing Test HAVE_CXX_FLAG_WSHADOW
-- Performing Test HAVE_CXX_FLAG_WSHADOW - Success
-- Performing Test HAVE_CXX_FLAG_WFLOAT_EQUAL
-- Performing Test HAVE_CXX_FLAG_WFLOAT_EQUAL - Success
-- Performing Test HAVE_CXX_FLAG_WERROR
-- Performing Test HAVE_CXX_FLAG_WERROR - Success
-- Performing Test HAVE_CXX_FLAG_WSUGGEST_OVERRIDE
-- Performing Test HAVE_CXX_FLAG_WSUGGEST_OVERRIDE - Success
-- Performing Test HAVE_CXX_FLAG_PEDANTIC
-- Performing Test HAVE_CXX_FLAG_PEDANTIC - Success
-- Performing Test HAVE_CXX_FLAG_PEDANTIC_ERRORS
-- Performing Test HAVE_CXX_FLAG_PEDANTIC_ERRORS - Success
-- Performing Test HAVE_CXX_FLAG_WSHORTEN_64_TO_32
-- Performing Test HAVE_CXX_FLAG_WSHORTEN_64_TO_32 - Failed
-- Performing Test HAVE_CXX_FLAG_FSTRICT_ALIASING
-- Performing Test HAVE_CXX_FLAG_FSTRICT_ALIASING - Success
-- Performing Test HAVE_CXX_FLAG_WNO_DEPRECATED_DECLARATIONS
-- Performing Test HAVE_CXX_FLAG_WNO_DEPRECATED_DECLARATIONS - Success
-- Performing Test HAVE_CXX_FLAG_WNO_DEPRECATED
-- Performing Test HAVE_CXX_FLAG_WNO_DEPRECATED - Success
-- Performing Test HAVE_CXX_FLAG_WSTRICT_ALIASING
-- Performing Test HAVE_CXX_FLAG_WSTRICT_ALIASING - Success
-- Performing Test HAVE_CXX_FLAG_WD654
-- Performing Test HAVE_CXX_FLAG_WD654 - Failed
-- Performing Test HAVE_CXX_FLAG_WTHREAD_SAFETY
-- Performing Test HAVE_CXX_FLAG_WTHREAD_SAFETY - Failed
-- Performing Test HAVE_CXX_FLAG_COVERAGE
-- Performing Test HAVE_CXX_FLAG_COVERAGE - Success
-- Compiling and running to test HAVE_STD_REGEX
-- Performing Test HAVE_STD_REGEX -- success
-- Compiling and running to test HAVE_GNU_POSIX_REGEX
-- Performing Test HAVE_GNU_POSIX_REGEX -- failed to compile: Change Dir: /home/bjoern/Documents/TestProject/build/CMakeFiles/CMakeTmp

Run Build Command(s):/usr/bin/gmake -f Makefile cmTC_8c987/fast && /usr/bin/gmake  -f CMakeFiles/cmTC_8c987.dir/build.make CMakeFiles/cmTC_8c987.dir/build
gmake[1]: Entering directory '/home/bjoern/Documents/TestProject/build/CMakeFiles/CMakeTmp'
Building CXX object CMakeFiles/cmTC_8c987.dir/gnu_posix_regex.cpp.o
/usr/lib64/ccache/c++   -std=c++11  -Wall  -Wextra  -Wshadow  -Wfloat-equal  -Wsuggest-override  -pedantic  -pedantic-errors  -fstrict-aliasing  -Wno-deprecated-declarations  -Wstrict-aliasing  -o CMakeFiles/cmTC_8c987.dir/gnu_posix_regex.cpp.o -c /home/bjoern/Documents/TestProject/build/_deps/googlebenchmark-src/cmake/gnu_posix_regex.cpp
/home/bjoern/Documents/TestProject/build/_deps/googlebenchmark-src/cmake/gnu_posix_regex.cpp:1:10: fatal error: gnuregex.h: No such file or directory
    1 | #include <gnuregex.h>
      |          ^~~~~~~~~~~~
compilation terminated.
gmake[1]: *** [CMakeFiles/cmTC_8c987.dir/build.make:78: CMakeFiles/cmTC_8c987.dir/gnu_posix_regex.cpp.o] Error 1
gmake[1]: Leaving directory '/home/bjoern/Documents/TestProject/build/CMakeFiles/CMakeTmp'
gmake: *** [Makefile:127: cmTC_8c987/fast] Error 2


-- Compiling and running to test HAVE_POSIX_REGEX
-- Performing Test HAVE_POSIX_REGEX -- success
-- Compiling and running to test HAVE_STEADY_CLOCK
-- Performing Test HAVE_STEADY_CLOCK -- success
-- Configuring done
-- Generating done
-- Build files have been written to: /home/bjoern/Documents/TestProject/build
```

**Comment by dominichamon** [comment]
yes it's fixed. this is output explaining whit HAVE_GNU_POSIX_REGEX isn't available, but it still generates the build files. you should be able to `make` the library at this point.
